client/src/js/services/CommonModuleState.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+/**
+ * @description
+ * Very simple service to define state variables that are common to most
+ * modules but are defined under different names and structures. This service is
+ * essentially a semantic wrapper around many common states.
+ *
+ * @returns {object} Getters and Setters for common modules states
+ *
+ * @todo Refactor API to remove redundant states
+ *
+ * @todo Using getters and setters to semantically expose variables may be more
+ * costly than it is worth, this should be investigated
+ */
+
+angular.module('bhima.services').factory('ModuleState', ModuleState);
+    
+ModuleState.$inject = ['$timeout'];
+
+function ModuleState ($timeout) { 
+  
+  function Session () { 
+
+    // Components relying on this should not assume invalid as default
+    this.__valid = undefined;
+    this.__success = undefined;
+
+    this.__complete = false;
+    this.__initialised = false;
+    this.__loading = false;
+
+    // Expose available properties
+    Object.defineProperty(this, 'isValid', { 
```

Comment:
For syntactic sugar, you could use `Object.defineProperties()` instead.  [Relevant documentation](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperties).